open_instruct/ppo_fast.py

```diff
@@ -0,0 +1,1866 @@
+
+    # Batch sizes
+    per_device_train_batch_size: int = 1
+    """The forward batch size per device (local_micro_batch_size)"""
+    total_episodes: int = 100000
+    """The total number of episodes in the dataset"""
+    world_size: Optional[int] = None
+    """RUNTIME VALUE: The number of processes (GPUs) to use"""
+    num_training_steps: Optional[int] = None
+    """RUNTIME VALUE: The number of training_steps to train"""
+    num_evals: int = 10
+    """The number of evaluations to run throughout training"""
+    eval_freq: Optional[int] = None
+    """RUNTIME VALUE: The frequency of evaluation steps"""
+    save_freq: int = -1
+    """How many train steps to save the model"""
+
+    # Generation
+    response_length: int = 256
+    """the length of the response"""
+    temperature: float = 0.7
+    """the sampling temperature"""
+    num_unique_prompts_rollout: int = 16
+    """The number of unique prompts during rollout"""
+    num_samples_per_prompt_rollout: int = 4
+    """the number of samples to generate per prompt during rollout, useful for easy-star"""
+    stop_strings: Optional[List[str]] = None
+    """List of strings that stop the generation when they are generated.
+    The returned output will not contain the stop strings."""
+
+    # Algorithm
+    async_mode: bool = True
+    """Whether to run the generation in async mode which learns from the second latest policy like Cleanba (https://arxiv.org/abs/2310.00036)"""
+    num_epochs: int = 1
+    """the number of epochs to train"""
+    num_mini_batches: int = 1
+    """Number of minibatches to split a batch into"""
+    value_num_mini_batches: int = 1
+    """Number of minibatches to split a batch into for the value function"""
+    beta: float = 0.05
+    """the beta value of the RLHF objective (KL coefficient)"""
+    cliprange: float = 0.2
+    """the clip range"""
+    cliprange_value: float = 0.2
+    """the clip range for the value function"""
+    gamma: float = 1.0
+    """the discount factor"""
+    lam: float = 1.0
+    """the lambda value for GAE"""
+    kl_estimator: Literal["kl1", "kl2", "kl3", "kl4"] = "kl3"
+    """the KL estimator to use"""
+    pack_length: int = 512
+    """the length of the pack (you should prob set to the max length of the model)"""
+    masked_mean_axis: Optional[int] = None
+    """the axis to compute the mean of the masked values"""
+    alpha: float = 0.6
+    """The alpha value for doing polyak updates (ref_param = alpha * param + (1 - alpha) * ref_param)
+    reference: [TR-DPO](https://huggingface.co/papers/2404.09656), but it's actually pretty commonly
+    used. E.g., [TD3](https://arxiv.org/abs/1802.09477) uses https://github.com/vwxyzjn/cleanrl/blob/dcc289fc6f0bda492fa7360a155262cf826b12a5/cleanrl/td3_continuous_action.py#L269
+    """
+    ref_policy_update_freq: Optional[int] = None
+    """How many training steps to take before updating the reference policy."""
+    reward_model_name_or_path: str = "EleutherAI/pythia-160m"
```

Comment:
I wonder if it would also be useful to be able to specify a value model path?

Reply:
value_model_name_or_path: str = "EleutherAI/pythia-160m"
    """the name or path to the value model"""
    value_model_revision: Optional[str] = None
    """the revision of the value model"""